### PR TITLE
fix bug: remove spark applications list cache

### LINKED ISSUE #739: Spark job view list
There're no newer applications after reloading the job view page.

### PR DESCRIPTION
**Pull request by voidfunction**
Fix #739

### CODE REVIEW

**Comment by voidfunction** [comment]
Build Failed. error message : 
```
Tests run: 13, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 41.283 sec <<< FAILURE! - in com.microsoft.tooling.msservices.serviceexplorer.azure.rediscache.RedisCacheCreateIntegrationTest
testRedisCacheCreateBasicWithExistedResGrp(com.microsoft.tooling.msservices.serviceexplorer.azure.rediscache.RedisCacheCreateIntegrationTest)  Time elapsed: 30.338 sec  <<< ERROR!
java.lang.Exception: Cannot create workMock instance
	at com.microsoft.tooling.IntegrationTestBase.setUpStep(IntegrationTestBase.java:113)
	at com.microsoft.tooling.msservices.serviceexplorer.azure.rediscache.RedisCacheCreateIntegrationTest.setUp(RedisCacheCreateIntegrationTest.java:132)

testRedisCacheCreateBasicWithExistedResGrp(com.microsoft.tooling.msservices.serviceexplorer.azure.rediscache.RedisCacheCreateIntegrationTest)  Time elapsed: 30.34 sec  <<< ERROR!
org.apache.http.conn.HttpHostConnectException: Connect to localhost:3013 [localhost/127.0.0.1] failed: Connection refused (Connection refused)
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at org.apache.http.conn.socket.PlainConnectionSocketFactory.connectSocket(PlainConnectionSocketFactory.java:74)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:141)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:353)
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:380)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:184)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:184)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:82)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:107)
	at com.github.tomakehurst.wiremock.client.HttpAdminClient.safelyExecuteRequest(HttpAdminClient.java:311)
	at com.github.tomakehurst.wiremock.client.HttpAdminClient.postJsonAssertOkAndReturnBody(HttpAdminClient.java:261)
	at com.github.tomakehurst.wiremock.client.HttpAdminClient.resetAll(HttpAdminClient.java:144)
	at com.github.tomakehurst.wiremock.client.WireMock.resetMappings(WireMock.java:210)
	at com.microsoft.tooling.IntegrationTestBase.resetTest(IntegrationTestBase.java:158)
	at com.microsoft.tooling.msservices.serviceexplorer.azure.rediscache.RedisCacheCreateIntegrationTest.tearDown(RedisCacheCreateIntegrationTest.java:200)
```

**Comment by wezhang** [comment]
@adashen hit RedisCacheCreateIntegrationTest issue again...

**Comment by Eskibear** [comment]
@wezhang Log reveals this time what you hit is not IntegrationTest issue but the network issue, both travis and jenkins. :(
The integration issue also blocks us, we're investigating now. 
```
org.apache.maven.InternalErrorException: Internal error: java.lang.RuntimeException: Failed to load p2 repository with ID 'mars' from location http://download.eclipse.org/releases/mars
```

**Comment by Eskibear** [comment]
retest this please

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/jobs/SparkJobHttpHandler.java [hunk 1]
```diff
@@ -48,9 +50,13 @@ public void handle(HttpExchange httpExchange) throws IOException {
...
             if (path.equalsIgnoreCase("/applications/") && requestDetail.getAppId().equalsIgnoreCase("0")) {
-                List<Application> applications = JobViewCacheManager.getSparkApplications(requestDetail.getCluster());
-                Optional<String> responseString = ObjectConvertUtils.convertObjectToJsonString(applications);
-                JobUtils.setResponse(httpExchange, responseString.orElseThrow(IOException::new));
+                try {
+                    List<Application> applications = SparkRestUtil.getSparkApplications(requestDetail.getCluster());
+                    Optional<String> responseString = ObjectConvertUtils.convertObjectToJsonString(applications);
+                    JobUtils.setResponse(httpExchange, responseString.orElseThrow(IOException::new));
+                } catch (HDIException e) {
+                    DefaultLoader.getUIHelper().logError("get applications list error", e);
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/jobs/SparkJobHttpHandler.java]: You can try to implement ILogger interface and use log() to use slf4j to manage the errors.

**Comment by voidfunction** [comment]
Fail again: 
```
[ERROR] Internal error: java.lang.RuntimeException: Failed to load p2 repository with ID 'mars' from location http://download.eclipse.org/releases/mars: Communication with repository at http://download.eclipse.org/releases/mars failed. Read timed out -> [Help 1]
org.apache.maven.InternalErrorException: Internal error: java.lang.RuntimeException: Failed to load p2 repository with ID 'mars' from location http://download.eclipse.org/releases/mars
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:167)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:216)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:160)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: java.lang.RuntimeException: Failed to load p2 repository with ID 'mars' from location http://download.eclipse.org/releases/mars
	at org.eclipse.tycho.p2.target.TargetPlatformFactoryImpl.loadMetadataRepository(TargetPlatformFactoryImpl.java:302)
	at org.eclipse.tycho.p2.target.TargetPlatformFactoryImpl.gatherExternalInstallableUnits(TargetPlatformFactoryImpl.java:270)
	at org.eclipse.tycho.p2.target.TargetPlatformFactoryImpl.createTargetPlatform(TargetPlatformFactoryImpl.java:176)
	at org.eclipse.tycho.p2.target.TargetPlatformFactoryImpl.createTargetPlatform(TargetPlatformFactoryImpl.java:128)
	at org.eclipse.tycho.p2.target.TargetPlatformFactoryImpl.createTargetPlatform(TargetPlatformFactoryImpl.java:1)
	at org.eclipse.tycho.p2.manager.ReactorRepositoryManagerImpl.computePreliminaryTargetPlatform(ReactorRepositoryManagerImpl.java:84)
	at org.eclipse.tycho.p2.resolver.P2DependencyResolver.computePreliminaryTargetPlatform(P2DependencyResolver.java:223)
	at org.eclipse.tycho.core.resolver.DefaultTychoResolver.resolveProject(DefaultTychoResolver.java:109)
	at org.eclipse.tycho.core.maven.TychoMavenLifecycleParticipant.afterProjectsRead(TychoMavenLifecycleParticipant.java:95)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:318)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:155)
	... 11 more
Caused by: org.eclipse.equinox.p2.core.ProvisionException: Communication with repository at http://download.eclipse.org/releases/mars failed.
	at org.eclipse.equinox.internal.p2.repository.CacheManager.createCache(CacheManager.java:202)
	at org.eclipse.tycho.p2.remote.RemoteRepositoryCacheManager.createCache(RemoteRepositoryCacheManager.java:
```

**Comment by voidfunction** [comment]
retest this please

**Comment by Eskibear** [comment]
retest this please

**Comment by voidfunction** [comment]
Failed again:
```
+mvn install -f /home/travis/build/Microsoft/azure-tools-for-java/PluginsAndFeatures/AddLibrary/AzureLibraries/pom.xml -Dmaven.repo.local=/home/travis/build/Microsoft/azure-tools-for-java/.repository -q
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=192m; support was removed in 8.0
+mvn clean install -f /home/travis/build/Microsoft/azure-tools-for-java/PluginsAndFeatures/azure-toolkit-for-eclipse/pom.xml -q
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=192m; support was removed in 8.0
[ERROR] Internal error: java.lang.RuntimeException: Failed to load p2 repository with ID 'mars' from location http://download.eclipse.org/releases/mars: Unable to read repository at http://download.eclipse.org/releases/mars. Unable to read repository at http://download.eclipse.org/releases/mars/201510021000/content.jar. connect timed out -> [Help 1]
org.apache.maven.InternalErrorException: Internal error: java.lang.RuntimeException: Failed to load p2 repository with ID 'mars' from location http://download.eclipse.org/releases/mars
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:167)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:216)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:160)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: java.lang.RuntimeException: Failed to load p2 repository with ID 'mars' from location http://download.eclipse.org/releases/mars
	at org.eclipse.tycho.p2.target.TargetPlatformFactoryImpl.loadMetadataRepository(TargetPlatformFactoryImpl.java:302)
	at org.eclipse.tycho.p2.target.TargetPlatformFactoryImpl.gatherExternalInstallableUnits(TargetPlatformFactoryImpl.java:270)
	at org.eclipse.tycho.p2.target.TargetPlatformFactoryImpl.createTargetPlatform(TargetPlatformFactoryImpl.java:176)
	at org.eclipse.tycho.p2.target.TargetPlatformFactoryImpl.createTargetPlatform(TargetPlatformFactoryImpl.java:128)
	at org.eclipse.tycho.p2.target.TargetPlatformFactoryImpl.createTargetPlatform(TargetPlatformFactoryImpl.java:1)
	at org.eclipse.tycho.p2.manager.ReactorRepositoryManagerImpl.computePreliminaryTargetPlatform(ReactorRepositoryManagerImpl.java:84)
	at org.eclipse.tycho.p2.resolver.P2DependencyResolver.computePreliminaryTargetPlatform(P2DependencyResolver.java:223)
	at org.eclipse.tycho.core.resolver.DefaultTychoResolver.resolveProject(DefaultTychoResolver.java:109)
	at org.eclipse.tycho.core.maven.TychoMavenLifecycleParticipant.afterProjectsRead(TychoMavenLifecycleParticipant.java:95)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:318)
```

**Comment by voidfunction** [comment]
@adashen @Eskibear Could you help merge this PR since the default build passed.

**Pull request merged**